tools/generate-helpers/pg-table-bindings/store.go.tpl

```diff
@@ -656,31 +686,6 @@ func (s *storeImpl) Walk(ctx context.Context, fn func(obj *{{.Type}}) error) err
 	return nil
 }
 
-{{ if .Obj.IsDirectlyScoped }}
-    func isInScope(obj *{{.Type}}, eas *effectiveaccessscope.ScopeTree) bool {
```

Comment:
Could the function limit its role to extracting the `clusterID` and `namespace`, then let the `eas` object perform the scope check ?
```
func isInScope(obj *{{.Type}}, eas effectiveAccessScope.ScopeTree) bool {
    clusterID := {{ "obj" | .Obj.GetClusterID }}
    {{ if .Obj.IsClusterScope -}}
    return eas.IsInScope(clusterIDKey)
    {{- else if .Obj.IsNamespaceScope -}}
    namespace := {{ "obj" | .Obj.GetNamcespace }}
    return eas.IsInScope(clusterID, namespace)
    {{- else -}}
    return false
    {{- end }}
}
```
This would require `IsInScope(keys ...ScopeKey)`/`isInScope(keys ...ScopeKey)` functions in `pkg/sac/effectiveaccessscope/effective_access_scope.go` the like of
```
func (root *ScopeTree) IsInScope(keys ...string) bool {
    if root == nil || root.State == Excluded { return false }
    if root.State == Included { return true }
    if len(keys) == 0 { return false }
    clusterID := keys[0].String()
    clusterSubTree := root.GetClusterByID(clusterID)
    if clusterSubTree == nil { return false }
    return clusterSubTree.isInScope(keys[1:])
}
```

Reply:
Created a task for it https://issues.redhat.com/browse/ROX-10615